### PR TITLE
Added time report page

### LINKED ISSUE #16: Reports generation
With the values from the form, the page needs to generate a report and display it in a presentable way.

Story points: 8

### PR DESCRIPTION
**Pull request by aaroncohen73**
Adds the URL endpoint "/report" that gets times from TimeSync and allows users to filter those times by entering parameters into a form.

Closes #16

### CODE REVIEW

File: app/templates/report.html [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 {% block title %} View Times {% endblock %}
 
+{% block imports %}
+<script type="text/javascript" src="{{ url_for('static', filename='scripts/tableSort.js') }}"></script>
```
Comment (subnomo) [inline app/templates/report.html]: This should go in the `scripts` block.

File: app/templates/report.html [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{% extends "layout.html" %}
+
+{% block title %} View Times {% endblock %}
+
+{% block imports %}
+<script type="text/javascript" src="{{ url_for('static', filename='scripts/tableSort.js') }}"></script>
```
Comment (subnomo) [inline app/templates/report.html:6]: This should go at the top of the `scripts` block.

File: app/templates/layout.html [hunk 3]
```diff
@@ -3,6 +3,7 @@
     <head>
         <title>{% block title %}{% endblock %}</title>
         {% block imports %}{% endblock %}
+        {% block scripts %}{% endblock %}
```
Comment (subnomo) [inline app/templates/layout.html]: This should go at the very end of the `<body>`. If you do this, you won't need `window.onload` to initialize your script.

File: app/views.py [hunk 4]
```diff
@@ -144,3 +144,49 @@ def submit():
...
+    if 'token' not in session:
+        return redirect(url_for('login', next=request.url_rule))
+
+    ts = pymesync.TimeSync(baseurl=app.config['TIMESYNC_URL'],
+                           test=app.config['TESTING'], token=session['token'])
+
+    form = forms.GenerateReportForm()
+
+    query = dict()
+
```
Comment (subnomo) [inline app/views.py]: Just a couple of comments in this function would be helpful.

File: app/views.py [hunk 5]
```diff
@@ -144,3 +144,49 @@ def submit():
...
+        flash("Invalid form input")
+
+    times = ts.get_times(query_parameters=query)
+
+    if 'error' in times or 'pymesync error' in times:
+        flash("Error")
+        flash(times)
+        times = list()
+
+    return render_template('report.html', form=form, times=times)
```
Comment (subnomo) [inline app/views.py]: Just a couple of comments in this function would be helpful.

File: app/views.py [hunk 6]
```diff
@@ -144,3 +144,49 @@ def submit():
...
+    ts = pymesync.TimeSync(baseurl=app.config['TIMESYNC_URL'],
+                           test=app.config['TESTING'], token=session['token'])
+
+    form = forms.GenerateReportForm()
+
+    query = dict()
+
+    if form.validate_on_submit():
+        req_form = request.form
+
```
Comment (subnomo) [inline app/views.py:167]: Just a couple of comments in this function would be helpful.

**Comment by Kennric** [comment]
I second @subnomo's comments - otherwise this looks ready to go.

**Comment by subnomo** [comment]
This also closes #15.

:+1:

File: app/__init__.py [hunk 7]
```diff
@@ -1,6 +1,7 @@
 from flask import Flask
 
-app = Flask(__name__)
+app = Flask(__name__, static_folder="static")
```
Comment (pop) [inline app/__init__.py:3]: `static_folder` should got in a config file and default to `"static"` if it us unset.

**Comment by pop** [comment]
+1

There's a lot of code here and I can't say I went through it with a fine-toothed comb. That said the code/tests look pretty good and the everything passes.

**Comment by matthewrsj** [comment]
General style comment regarding function and variable naming in python: https://www.python.org/dev/peps/pep-0008/#function-names

Not overly relevant to this PR but nice to keep in mind.

**Pull request merged**